src/main/java/uk/gov/companieshouse/reconciliation/service/elasticsearch/ElasticsearchTransformer.java

```diff
@@ -5,29 +5,27 @@
 import org.elasticsearch.search.SearchHit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.stereotype.Component;
 import uk.gov.companieshouse.reconciliation.model.ResultModel;
 import uk.gov.companieshouse.reconciliation.model.Results;
+import uk.gov.companieshouse.reconciliation.service.elasticsearch.primary.ElasticsearchPrimaryIndexTransformer;
 
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.Iterator;
 import java.util.List;
-import java.util.Map;
-import java.util.Optional;
 
 /**
  * Transform {@link SearchHit search hits} retrieved from an Elasticsearch index into a collection
  * of {@link Results results}.
  */
-@Component
-public class ElasticsearchTransformer {
+public abstract class ElasticsearchTransformer {
```

Comment:
Let's use composition instead of inheritance to make it easier to change the transform method's parameter list later on.